webui/react/src/services/apiBuilder.ts

```diff
@@ -54,3 +59,30 @@ export function generateApi<Input, Output>(api: Api<Input, Output>) {
     }
   };
 }
+
+/*
+  consumeStream is used to consume streams from the generated TS client.
+  We use the provided fetchParamCreator to create fetch arguments and use that
+  to make a request and handle events one by one.
+  Example:
+  consumeStream<DetSwagger.V1TrialLogsResponse>(
+    DetSwagger.ExperimentsApiFetchParamCreator().determinedTrialLogs(1),
+    console.log,
+  ).then(() => console.log('finished'));
+*/
+export const consumeStream = async <T = unknown>(
+  fetchArgs: DetSwagger.FetchArgs, onEvent: (event: T) => void): Promise<void> => {
+  let response;
```

Comment:
chore: this declaration can be moved into the `try` block and be a `const response = await...`